crates/extensions/c8y_firmware_manager/src/actor.rs

```diff
@@ -124,6 +125,7 @@ impl FirmwareManagerActor {
             input_receiver,
             worker: FirmwareManagerWorker {
                 config: Arc::new(config),
+                executing: false,
```

Comment:
This has been introduced to keep the current behavior unchanged,
as tested by `handle_request_child_device_with_failed_download`.

However, I think the current behavior is weird. The executing status is sent to c8y when the child device starts the operation (and not when the mapper starts the download), and there is a single exception to this rule: when the download fails, an executing status is sent to c8y by the mapper.

It would be better to send this status when the mapper starts to process the request by looking in its cache for the firmware. However, doing so would require all the tests to be updated (as they are sensitive to message order). Something, I really want to avoid in a refactoring PR.